File: app/src/main/java/com/abdelmeged/ahmed/networkitotest/java/NormalActivity.java
package com.abdelmeged.ahmed.networkitotest.java;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import com.abdelmeged.ahmed.networkitotest.R;
import com.ahmedabdelmeged.networkito.ConnectionType;
import com.ahmedabdelmeged.networkito.ConnectivityChangeListener;
import com.ahmedabdelmeged.networkito.Networkito;


public class NormalActivity extends AppCompatActivity {

    private Networkito networkito;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        networkito = new Networkito(this);

        if (networkito.isAvailableInternetConnection()) {
            Toast.makeText(this, "Available Internet Connection", Toast.LENGTH_SHORT).show();
        }

        networkito.setConnectivityChangeListener(new ConnectivityChangeListener() {
            @Override
            public void onInternetConnected(ConnectionType connectionType) {
                Toast.makeText(NormalActivity.this, "connected: " + connectionType.name(), Toast.LENGTH_SHORT).show();
            }

            @Override
            public void onInternetDisconnected() {
                Toast.makeText(NormalActivity.this, "disconnected", Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    protected void onStop() {
        super.onStop();
        networkito.unRegisterNetworkitoReceiver();
    }
}

File: app/src/main/java/com/abdelmeged/ahmed/networkitotest/kotlin/NormalActivity.kt
package com.abdelmeged.ahmed.networkitotest.kotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import com.abdelmeged.ahmed.networkitotest.R
import com.ahmedabdelmeged.networkito.ConnectionType
import com.ahmedabdelmeged.networkito.ConnectivityChangeListener
import com.ahmedabdelmeged.networkito.Networkito
import org.jetbrains.anko.toast

class NormalActivity : AppCompatActivity() {

    lateinit var networkito: Networkito

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        networkito = Networkito(this)

        if (networkito.isAvailableInternetConnection) {
            toast("Available Internet Connection")
        }

        networkito.setConnectivityChangeListener(object : ConnectivityChangeListener {
            override fun onInternetConnected(connectionType: ConnectionType?) {
                toast("connected: " + connectionType!!.name)
            }

            override fun onInternetDisconnected() {
                toast("disconnected")
            }

        })
    }


    override fun onStop() {
        super.onStop()
        networkito.unRegisterNetworkitoReceiver()
    }
}
